### ISSUE TITLE
[Bug]: Unable to load joint trajectory controller for Kinova Gen3

### ISSUE THREAD

**Issue by sbtc-sipbb** [aica-technology/api#142]
### Image manifest

```Shell
#syntax=ghcr.io/aica-technology/app-builder:v1

[packages]
"@aica/base" = "v3.1.1"

# add components
"@aica/components/signal-processing" = "v1.0.1"
"@aica/components/auto-frame-calibration" = "v1.0.0"
"@aica/components/pose-detection" = "v1.0.2"
"@aica/components/task-optimization" = "v3.0.0"

# add hardware collections 
"@aica/collections/ur-collection" = "v3.0.4"
"@aica/collections/kinova-gen3-collection" = "v0.1.1"
```

### Brief description

I tried to replicate the example of Dominic from last Wednesday (commanding the robot through AICA -> Ros2Control joint trajectory controller -> position interface). But apparently the trajectory controller cannot be activated.   

### Steps to reproduce

Load the configuration file provided by Dominic and change the urdf to a custom version of the one from kinova-gen3-collection:
```yaml
# yaml-language-server: $schema=https://docs.aica.tech/schemas/1-2-0/application.schema.json

on_start:
  load:
    - { hardware: robot }
    - { hardware: robot, controller: broadcaster }
    - { hardware: robot, controller: trajectory }
  switch_controllers:
    - { hardware: robot, activate: [ broadcaster, trajectory ] }

hardware:
  robot:
    # Here I changed to a custom URDF based on the kinova-gen3-collection 
    # because this one is missing and the standard one from the kinova-gen3- 
    # collection has the wrong IP
    urdf: /home/ros2/examples/urdf/kinova_gen3.urdf 
    rate: 1000
    controllers:
      broadcaster:
        plugin: modulo_controllers/RobotStateBroadcaster
      trajectory:
        plugin: joint_trajectory_controller/JointTrajectoryController
        parameters:
          command_interfaces: [position]
          state_interfaces: [position]
```

As soon as I click on "Generate Graph", the joint_trajectory_controller disappears. The resulting file as copied from the AICA interface is then:
```yaml
on_start:
  load:
    - hardware: robot
    - hardware: robot
      controller: broadcaster
    - hardware: robot
      controller: trajectory
  switch_controllers:
    - hardware: robot
      activate:
        - broadcaster
        - trajectory
components: {}
hardware:
  robot:
    position:
      x: 500
      y: 0
    urdf: kinova_gen3_sbtc
    rate: 1000
    controllers:
      broadcaster:
        plugin: modulo_controllers/RobotStateBroadcaster
```

On top of that, I get the error (see log output)

### Relevant log output

```Shell
[INFO] [1711352926.223158380] [StateEventHandler]: Handling event SwitchControllersEvent called by on_start
[INFO] [1711352926.223163554] [StateEventHandler]: Switching controllers on hardware interface robot: activate [ broadcaster trajectory ], deactivate [ ]
[WARN] [1711352926.223171437] [robot.ControllerManager]: Could not 'activate' controller with name 'trajectory' because no controller with this name exists
[ERROR] [1711352926.223179238] [robot.ControllerManager]: Aborting, no controller is switched! ('STRICT' switch)
[DEBUG] [1711352926.223638798] [KortexMultiInterfaceHardware]: No controller active in LOW_LEVEL_SERVOING mode !
[DEBUG] [1711352926.224671070] [KortexMultiInterfaceHardware]: No controller active in LOW_LEVEL_SERVOING mode !
```

### Supporting information

Here is the urdf that I used (btw, would be nice to have support for file uploads in urdf format):
[kinova_gen3_sbtc.zip](https://github.com/aica-technology/api/files/14740465/kinova_gen3_sbtc.zip)

**Comment by eeberhard** [comment]
Thank you very much for the bug report! 

It is indeed possible to use standard ros2_control control plugins such as the `joint_trajectory_controller/JointTrajectoryController` with the application. However, the problem as you noticed is that the graph editor mysteriously deletes this controller from the application code.

That is because currently the graph UI can only render certain "AICA" controllers because we wrote some meta-data to define what parameters, inputs and outputs these controllers have. Because any other controllers are not rendered in the UI, they then get removed from the code.

**We will work on a fix to keep all controllers in the code**, even if they cannot fully be rendered as AICA controllers. That way, you will still be able to run your application.

In the meantime, instead of using the graph editor UI, you can still run this application through the API. You just need to set the application and then start it. The easiest way to do it is through the Python client, available [here](https://pypi.org/project/aica-api/).

Thank you for your feedback and patience as we make a patch for this.

**Comment by domire8** [comment]
Also for future reference, you can see [here](https://docs.aica.tech/docs/reference/yaml-syntax#parameters-1) how you can avoid making a duplicate of the URDF if you only need to change a parameter that is under the `<hardware>` tag in the ros2 control urdf macro.

**Comment by eeberhard** [comment]
Hi @sbtc-sipbb, I'm happy to say this issue has been addressed in the newest v3.2.0 minor update today. You can upgrade your image manifest to use the version `"@aica/base" = "v3.2.0"`.

With this new fix, you can still load standard "third-party" (i.e. non-AICA) control plugins and components into the developer interface. Note that, as before, you won't see the parameters, inputs and outputs for these external controllers or components in the UI (because they don't have the AICA-specific interface descriptions). But, you will be able to load, activate, deactivate and unload the controllers with state events as normal.

I will mark this issue as resolved, but please do not hesitate to leave further feedback if you still experience problems related to this issue. Future bug reports and feature requests are also welcome.
